content/fr/articles/js/es2015/defaults.md

```diff
@@ -0,0 +1,132 @@
+---
+  date: "2015-12-03"
+  title: "ES6, ES2015 : valeurs par défaut des arguments"
+tags:
+  - javascript
+  - ES6
+  - ES2015
+authors:
+  - naholyr
+---
+
+Fatigués de taper ``if (typeof x === 'undefined') x = defaultValue`` ? Là encore
+ES6 va nous apporter une solution élégante.
+
+# Valeurs par défaut
+
+L'opérateur ``=`` permet maintenant en plus d'affecter une valeur à une variable
+de définir une valeur par défaut à un paramètre.
+
+## Paramètres de fonction
+
+Il est maintenant possible de spécifier une valeur par défaut à un paramètre de
+fonction. Cette valeur sera utilisée si le paramètre n'est pas fourni, ou qu'il
+est explicitement défini à *undefined*.
+
+```js
+function incr (value, step = 1) {
+  return value + step
+}
+
+incr(41) // 42
+incr(41, undefined) // 42
+incr(33, 9) // 42
+```
+
+### Valeurs par défaut dynamiques
+
+On peut spécifier une expression en tant que valeur par défaut. Cette expression
+sera évaluée **à l'appel de la fonction**, pas lors de sa déclaration.
+
+```js
+let defaultWho = 'world!'
+function hello (who = defaultWho.toUpperCase()) {
+  return 'Hello ' + who
+}
+
+hello() // 'Hello WORLD!'
+defaultWho = 'Anyone?'
+hello() // 'Hello ANYONE?'
+```
+
+### Réutilisation des paramètres précédents
+
+Dans l'expression d'une valeur par défaut, on peut réutiliser les paramètres
+**précédents** de la fonction :
+
+```js
+function foo (x = 1, y = x+1) {
```

Comment:
I wanted to "group" the value, hence the absence of space, but I agree it feels weird ^^